packages/db/src/rules.ts

```diff
@@ -0,0 +1,62 @@
+import Knex from 'knex';
+import { RuleRecord } from '@cumulus/types/api/rules';
+import { PostgresRule, PostgresCollectionRecord, PostgresProviderRecord } from './types';
+import { getRecordCumulusId } from './database';
+import { tableNames } from './tables';
+
+const getCollectionCumulusId = async (record: RuleRecord, dbClient: Knex) => {
+  if (record.collection) {
+    await getRecordCumulusId<PostgresCollectionRecord>(
+      { name: record.collection.name, version: record.collection.version },
+      tableNames.collections,
+      dbClient
+    );
+  }
+  return undefined;
+};
+
+const getProviderCumulusId = async (record: RuleRecord, dbClient: Knex) => {
+  if (record.provider) {
+    const result = await getRecordCumulusId<PostgresProviderRecord>(
+      { name: record.provider },
+      tableNames.providers,
+      dbClient
+    );
+    return result;
+  }
+  return undefined;
+};
+
+/**
+ * Generate a Postgres rule record from a DynamoDB record.
+ * @param {Object} record - A rule
+ * @param {Object} dbClient - Knex client for reading from RDS database
+ * @returns {Object} A rule record
+ */
+export const translateApiRuleToPostgresRule = async (
+  record: RuleRecord,
+  dbClient: Knex
+): Promise<PostgresRule> => {
+  const collectionCumulusId = await getCollectionCumulusId(record, dbClient);
+  const providerCumulusId = await getProviderCumulusId(record, dbClient);
+  const ruleRecord: PostgresRule = {
+    name: record.name,
+    workflow: record.workflow,
+    provider_cumulus_id: record.provider ? providerCumulusId : undefined,
```

Comment:
Since we already have a ternary expression here, we can simplify the code to:

```suggestion
    provider_cumulus_id: record.provider ? await getRecordCumulusId<PostgresProviderRecord>(
      { name: record.provider },
      tableNames.providers,
      dbClient
    ) : undefined,
```

And remove the `getProviderCumulusId` helper. And same thing for collections

Reply:
This was resolved but I don't see the changes?

Reply:
Hmm odd, I am not sure why the changes are there. Let me add them again.

Reply:
[`ea9c778` (#2012)](https://github.com/nasa/cumulus/pull/2012/commits/ea9c7784b4d0e9841c8ebee053f05bbdcce89ce7)